jobbergate-cli/jobbergate_cli/subapps/applications/app.py

```diff
@@ -0,0 +1,385 @@
+    :param: application_id:   The ``id`` of the Application to be uploaded
+    """
+
+    # Make static type checkers happy
+    assert jg_ctx.client is not None
+
+    with tempfile.TemporaryDirectory() as temp_dir_str:
+        build_path = pathlib.Path(temp_dir_str)
+        logger.debug("Building application tar file at {build_path}")
+        tar_path = build_application_tarball(application_path, build_path)
+        response_code = cast(
+            int,
+            make_request(
+                jg_ctx.client,
+                f"/applications/{application_id}/upload",
+                "POST",
+                expect_response=False,
+                abort_message="Request to upload application files was not accepted by the API",
+                support=True,
+                files=dict(upload_file=open(tar_path, "rb")),
+            ),
+        )
+        return response_code == 201
+
+
+@app.command()
+@handle_abort
+def create(
+    ctx: typer.Context,
+    name: str = typer.Option(
+        ...,
+        help="The name of the application to create",
+    ),
+    identifier: Optional[str] = typer.Option(
+        None,
+        help=f"The human-friendly identifier of the application. {IDENTIFIER_NOTE}",
+    ),
+    application_path: pathlib.Path = typer.Option(
+        ...,
+        help="The path to the directory where the application files are located",
+    ),
+    application_desc: Optional[str] = typer.Option(
+        None,
+        help="A helpful description of the application",
+    ),
+):
+    """
+    Create a new application.
+    """
+    req_data = load_default_config()
+    req_data["application_name"] = name
+    if identifier:
+        req_data["application_identifier"] = identifier
+
+    if application_desc:
+        req_data["application_description"] = application_desc
+
+    validate_application_files(application_path)
+
+    req_data["application_config"] = dump_full_config(application_path)
+    req_data["application_file"] = read_application_module(application_path)
+
+    jg_ctx: JobbergateContext = ctx.obj
+
+    # Make static type checkers happy
+    assert jg_ctx.client is not None
+
+    result = cast(
+        Dict[str, Any],
+        make_request(
+            jg_ctx.client,
+            "/applications",
+            "POST",
+            expected_status=201,
+            abort_message="Request to create application was not accepted by the API",
+            support=True,
+            json=req_data,
+        ),
+    )
+    application_id = result["id"]
+
+    successful_upload = upload_application(jg_ctx, application_path, application_id)
+    if not successful_upload:
+        terminal_message(
+            """
+            The zipped application files could not be uploaded.
+
+            Try running the `update` command including the application path to re-upload.
+
+            [yellow]If the problem persists, please contact [bold]Omnivector <info@omnivector.solutions>[/bold]
```

Comment:
Added closing in commit 2a41204.

In this case, it wouldn't have mattered because the string ends before the closing happens. This is perfectly legal with Rich. However, I added the closing tag to be thorough.

I also used the `OV_CONTACT` constant instead.

Finally, the `color` flag to the `terminal_message` function only sets the color of the "subject" text.